include/nbl/asset/IMeshBuffer.h

```diff
@@ -14,214 +14,281 @@ namespace asset
-    }
-    inline uint64_t getAttribCombinedOffset(uint32_t attrId) const
-    {
-        const auto& buf = getAttribBoundBuffer(attrId);
-        return buf.offset+static_cast<uint64_t>(getAttribOffset(attrId));
-    }
-
-    //
-    inline const SBufferBinding<const BufferType>* getVertexBufferBindings() const
-    {
-        return reinterpret_cast<const SBufferBinding<const BufferType>*>(m_vertexBufferBindings);
-    }
-    inline const SBufferBinding<const BufferType>& getIndexBufferBinding() const
-    {
-        return reinterpret_cast<const SBufferBinding<const BufferType>&>(m_indexBufferBinding);
-    }
-    inline const PipelineType* getPipeline() const
-    {
-        return m_pipeline.get();
-    }
-    inline const DescSetType* getAttachedDescriptorSet() const
-    {
-        return m_descriptorSet.get();
-    }
-
-	//! Get type of index data which is stored in this meshbuffer.
-	/** \return Index type of this buffer. */
-	inline const E_INDEX_TYPE& getIndexType() const {return indexType;}
-	inline void setIndexType(const E_INDEX_TYPE& type)
-	{
-		indexType = type;
-	}
-
-	//! Get amount of indices in this meshbuffer.
-	/** \return Number of indices in this buffer. */
-	inline const uint64_t& getIndexCount() const {return indexCount;}
-	//! It sets amount of indices - value that is being passed to glDrawArrays as vertices amount or to glDrawElements as index amount.
-	/** @returns Whether set amount exceeds mapped buffer's size. Regardless of result the amount is set. */
-	inline bool setIndexCount(const uint64_t &newIndexCount)
-	{
-		/*
-		#ifdef _NBL_DEBUG
-			if (size<0x7fffffffffffffffuLL&&ixbuf&&(ixbuf->getSize()>size+offset))
-			{
-				os::Printer::log("MeshBuffer map vertex buffer overflow!\n",ELL_ERROR);
-				return;
-			}
-		#endif // _NBL_DEBUG
-		*/
-        indexCount = newIndexCount;
-        if (m_indexBufferBinding.buffer)
-        {
-            switch (indexType)
+    public:
+        _NBL_STATIC_INLINE_CONSTEXPR size_t MAX_PUSH_CONSTANT_BYTESIZE = 128u;
+
+        _NBL_STATIC_INLINE_CONSTEXPR size_t MAX_VERTEX_ATTRIB_COUNT = SVertexInputParams::MAX_VERTEX_ATTRIB_COUNT;
+        _NBL_STATIC_INLINE_CONSTEXPR size_t MAX_ATTR_BUF_BINDING_COUNT = SVertexInputParams::MAX_ATTR_BUF_BINDING_COUNT;
+
+    protected:
+        virtual ~IMeshBuffer() = default;
+
+        alignas(32) core::aabbox3df boundingBox;
+
+        SBufferBinding<BufferType> m_vertexBufferBindings[MAX_ATTR_BUF_BINDING_COUNT];
+        SBufferBinding<BufferType> m_indexBufferBinding;
+
+        //! Skin
+        SBufferBinding<BufferType> m_inverseBindPoseBufferBinding,m_jointAABBBufferBinding;
+        core::smart_refctd_ptr<SkeletonType> m_skeleton;
+
+        //! Descriptor set which goes to set=3
+        core::smart_refctd_ptr<DescSetType> m_descriptorSet;
+
+        alignas(64) uint8_t m_pushConstantsData[MAX_PUSH_CONSTANT_BYTESIZE]{};//by putting m_pushConstantsData here, alignas(64) takes no extra place
```

Comment:
`by putting m_pushConstantsData here, alignas(64) takes no extra place`
still true?

Reply:
according to my calculation yes, but feel free do doublecheck

Reply:
checkd it with debugger, its true